src/main/webapp/resources/js/pages/launch/parameters/SavedParameters.jsx

```diff
@@ -1,72 +1,168 @@
+    }
+  }
+
+  async function saveParameters() {
+    const fieldsValue = form.getFieldsValue(
+      sets[0].parameters.map((parameter) => parameter.name)
+    );
+    try {
+      const id = await saveModifiedParametersAs(
+        dispatch,
+        saveParamsForm.getFieldValue("name"),
+        fieldsValue
+      );
+      setCurrentSetId(id);
+      setModified(false);
+    } catch (e) {}
+  }
+
+  return (
     <>
-      <Form.Item label={i18n("SavedParameters.title")}>
+      <div>
+        <Form.Item
+          label={i18n("SavedParameters.title")}
+          tooltip={i18n("SavedParameters.tooltip")}
+          help={
+            Object.keys(modified).length ? (
+              <Alert
+                showIcon
+                style={{ marginBottom: SPACE_XS, marginTop: SPACE_XS }}
+                type={"warning"}
+                message={i18n("SavedParameters.modified")}
+                description={i18n("SavedParameters.modified.description")}
+                action={
+                  <Space>
+                    <Popover
+                      placement="bottomRight"
+                      trigger="click"
+                      content={
+                        <>
+                          <Typography.Text>
+                            {i18n("SavedParameters.modified.name")}
+                          </Typography.Text>
+                          <Form
+                            form={saveParamsForm}
+                            layout="inline"
+                            style={{
+                              width: 305,
+                            }}
+                          >
+                            <Form.Item name="name" required>
+                              <Input />
+                            </Form.Item>
+                            <Form.Item>
+                              <Button onClick={saveParameters}>
+                                {i18n("SavedParameters.modified.save")}
+                              </Button>
+                            </Form.Item>
+                          </Form>
+                        </>
+                      }
+                    >
+                      <Button size="small" type="ghost">
+                        {i18n("SavedParameters.modified.saveAs")}
+                      </Button>
+                    </Popover>
+                  </Space>
+                }
+              />
+            ) : null
+          }
+        >
+          <Select
+            style={{ width: `100%` }}
+            value={currentSetId}
+            onChange={(id) => updateSelectedSet(id)}
+            disabled={sets.length < 2}
+          >
+            {sets.map((set) => (
+              <Select.Option key={set.key} value={set.id}>
+                {set.label}
+              </Select.Option>
+            ))}
+          </Select>
+        </Form.Item>
+      </div>
+      {sets[currentSetId].parameters.map(({ name, label, value }) => (
```

Comment:
Add check to see if `sets[currentSetId]` exists before getting parameters from it.

Reply:
As per our discussion this is taken care of in the testing branch